### PR TITLE
ReadMe Revamp

### PR DESCRIPTION
**Pull request by Gregory711**
Revamped the README.md file to remove old sections, reorganize existing ones, add emojis, and add new roadmap section. Also, added logo, title, link to dev site, and description of Sudokuru broadly and Frontend specifically.

## Checklist for completing pull request:
- [ ] Update Changelog.json if any functionality has been changed for the end user.
- [ ] Test functionality on Mobile, Web, and Desktop
- [ ] Verify that any tests for new functionality are created and functional.
- [ ] If needed, update the Readme

## Summary by CodeRabbit

- **Documentation**
	- Enhanced the README.md with a clearer project overview, including purpose, goals, and features.
	- Added sections for "General Setup," "Mobile Setup," and "Running the Application Locally" to improve navigability.
	- Refined setup instructions and emphasized end-to-end testing using Playwright.
	- Expanded the roadmap section with details on project management and task categorization.

### CODE REVIEW

**Comment by Gregory711** [comment]
@coderabbitai review

File: README.md [hunk 1]
```diff
@@ -6,62 +12,59 @@
...
+- 📅 _Upcoming_: drills which let you practice individual strategies also powered by the Sudokuru npm library module
 
-For iOS, scan the QR code with your camera app, for Android you will need to scan the code from within the Expo app.
+# 🛠️ Local development
 
-For Web, hit the w key to start up the website at `localhost:19000`
+## ⚙️ General Setup:
 
-# Only use single puzzle for debugging
+1. Git clone this repository
```
Comment (Thomas-Gallant) [inline README.md:30]: Might be better to replace with the git command:

1. Run `git clone https://github.com/Sudokuru/Frontend`

File: README.md [hunk 2]
```diff
@@ -6,62 +12,59 @@
...
+2. Install Node.js `v19.5.0` or later
+3. Install npm `9.3.1` or later
+4. Run `npm i` in this repositories root folder
```
Comment (Thomas-Gallant) [inline README.md:33]: suggesting:
Run `npm i` in the root folder

File: README.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+![Sudokuru Logo](https://sudokuru.s3.amazonaws.com/goldLogoText.png)
+
+# The Official Cross-Platform Frontend for our Free Open Source Sudoku Project
+
+# 💻 Try our DEV site at: https://sudokuru.pages.dev/
+
```
Comment (Thomas-Gallant) [inline README.md:6]: Would be nice to link to our itch.io page for 
Android, Linux, and Windows builds https://sudokuru.itch.io/sudokuru

File: README.md [hunk 4]
```diff
@@ -6,62 +12,59 @@
 [![Codacy Badge](https://app.codacy.com/project/badge/Grade/1342f842a14f40cca856d6e81204f8ac)](https://app.codacy.com/gh/Sudokuru/Frontend/dashboard?utm_source=gh&utm_medium=referral&utm_content=&utm_campaign=Badge_grade)
 [![Maintainability](https://api.codeclimate.com/v1/badges/4cd1d1027b7b2532c680/maintainability)](https://codeclimate.com/github/Sudokuru/Frontend/maintainability)
 
-# Start the frontend app
-
-Run `npm i` in the root folder  
-Run `npm run start`
+## Sudokuru is an open-source project focused on developing a world-class, cross-platform Sudoku app. We aim to provide a delightful user experience while also contributing to the community by building a collection of reusable software modules. These modules are designed to be free, well-documented, modern, and interoperable, allowing developers to easily incorporate them into their own Sudoku-related projects.
```
Comment (Thomas-Gallant) [inline README.md:16]: Could add a "Supported Platforms" section

### Supported Platforms

####  :white_check_mark: Tier 1 Support
- Web

#### :shipit: Teir 2 Support
- Android
- Windows
- Linux

#### :running: Planned Support
- iOS
- MacOS

#### Legend

 :white_check_mark: Tier 1 Support - We have full end to end test coverage for all new and existing functionality. We do manual testing of new features before deployment. 

:shipit: Teir 2 Support - We have partial or zero end to end test coverage for new and existing functionality. We may not always do manual testing of new features before deployment. 

:running: Planned Support - We do not currently support this platform, but we plan on supporting in the future.

File: README.md [hunk 5]
```diff
@@ -6,62 +12,59 @@
...
 
-For testing with a mobile device you can download the Expo app.  
-You will also need to be logged in to Expo on the development device (laptop/desktop that is running the code).  
-To login type: `expo login` and fill out the parameters.
+### This frontend module is the primary user interface for Sudokuru, providing a cross-platform Sudoku experience built with React Native Web and TypeScript. It offers a clean, intuitive design with features such as:
 
-Contact Thomas to get added to the Expo organization so that you can log in with your own email and password.
+- 🎮 Play Sudoku across 9 different difficulty levels generated by the Sudokuru Clearinghouse module
+- 🎓 Learn how to play Sudoku with lessons from the basics all the way to advanced strategies
```
Comment (Thomas-Gallant) [inline README.md:20]: Maybe rephrase this to:
Sudoku lessons from basic to advanced strategies

File: README.md [hunk 6]
```diff
@@ -6,62 +12,59 @@
...
+- 📊 Statistics to track your progress
+- ⚙️ Various preferences including ones to customize dynamic board highlighting
```
Comment (Thomas-Gallant) [inline README.md:22]: I think we can be a bit more vague here and say "Sensible default settings for casual players with options to customize the playing experience for users with different playstyles."

File: README.md [hunk 7]
```diff
@@ -6,62 +12,59 @@
...
-[EAS Update Description](https://docs.expo.dev/eas-update/how-it-works/)
+## 🏃‍♂️ Running the Application Locally
 
-To make use of EAS update, run following commands:
+1. Run `npm run start` from this repositories root folder
+2. For iOS, scan the QR code with your camera app, for Android you will need to scan the code from within the Expo app.
+3. For Web, hit the w key to start up the website at `localhost:19000`
 
-1. `npm i -g eas-cli`
+## ✅ Playwright E2E Tests
```
Comment (Thomas-Gallant) [inline README.md:47]: 🎭 Playwright E2E tests
Lets use the playwright emoji

File: README.md (hunk 1 above)
Comment (Gregory711) [reply README.md:30]: Since some people use https and others use ssh urls for git clone going to keep it the way it is.

File: README.md (hunk 5 above)
Comment (Thomas-Gallant) [reply README.md:22]: or maybe "Learn how to play Sudoku with lessons for simple and advanced strategies"

**Review by Thomas-Gallant — APPROVED** [review]

**Pull request merged**